storage/ops.go

```diff
@@ -1487,6 +1488,139 @@ func (a ByBDName) Len() int           { return len(a) }
 func (a ByBDName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a ByBDName) Less(i, j int) bool { return a[i].Name < a[j].Name }
 
+// ValidateAdvancedPartitions returns an array of validation error
+// strings for the advanced partitions
+func ValidateAdvancedPartitions(medias []*BlockDevice) []string {
+	results := []string{}
+
+	bootFound := false
+	swapFound := false
+	rootFound := false
+
+	for _, curr := range medias {
+		for _, ch := range curr.Children {
+			clrFound := false
+			label := ch.PartitionLabel
+
+			if label != "" {
+				log.Debug("ValidateAdvancedPartitions: Found partition %s with label %s", ch.Name, label)
+			}
+
+			for _, part := range strings.Split(label, "_") {
+				lowerPart := strings.ToLower(part)
+
+				if !clrFound {
+					if lowerPart == "clr" {
+						clrFound = true
+					}
+					continue
+				}
+
+				switch lowerPart {
+				case "boot":
+					if bootFound {
+						warning := fmt.Sprintf("Found multiple %s partition labels", "CLR_BOOT")
+						results = append(results, warning)
+						log.Warning("ValidateAdvancedPartitions: %s %+v", warning, ch)
+					} else {
+						bootFound = true
+						if ch.FsType != "vfat" {
+							warning := fmt.Sprintf("%s must be %s", "CLR_BOOT", "vfat")
+							results = append(results, warning)
+							log.Warning("ValidateAdvancedPartitions: %s %+v", warning, ch)
+						}
+					}
+				case "root":
+					if rootFound {
+						warning := fmt.Sprintf("Found multiple %s partition labels", "CLR_ROOT")
+						results = append(results, warning)
+						log.Warning("ValidateAdvancedPartitions: %s %+v", warning, ch)
+					} else {
+						rootFound = true
+						if !(ch.FsType == "ext2" || ch.FsType == "ext3" ||
+							ch.FsType == "ext4" || ch.FsType == "xfs") {
+							warning := fmt.Sprintf("%s must be %s", "CLR_ROOT", "ext*|xfs")
+							results = append(results, warning)
+							log.Warning("ValidateAdvancedPartitions: %s %+v", warning, ch)
+						}
+					}
+				case "swap":
+					if swapFound {
+						warning := fmt.Sprintf("Found multiple %s partition labels", "CLR_SWAP")
+						results = append(results, warning)
+						log.Warning("ValidateAdvancedPartitions: %s %+v", warning, ch)
+					} else {
+						swapFound = true
+					}
+				case "e":
```

Comment:
Is it intentional to exclude case "f":?

Reply:
Yes, in this case we are validating the Advanced settings based on the labels.
Currently, there is no validation work if they specific "f" as it just forces the formatting of the file system.